### PR TITLE
General structure in Geometry group

### PR DESCRIPTION
**Pull request by mettelilienthal**
## Describe your changes

created initial content for general structure in Geometry group

## Mention a member

@ACMackenthun please review and check whether we can write complete sentences

### CODE REVIEW

File: content/geometry/general-structure/General_structure.adoc [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+The unit used for object dimensions are meters.
+
+== Coordinate systems
+The local coordinate system is a right-handed coordinate system according to ISO 8855 with the axes pointing to the following directions. For a non-rotated coordinate system the following applies:
+
+* Forward matches x-axis
+* Left matches y-axis
+* Up matches z-axis
+
+The coordinate system is in line with ASAM OpenDRIVE, ASAM OpenSCENARIO, ASAM OSI.
```
Comment (mettelilienthal) [inline content/geometry/general-structure/General_structure.adoc:13]:
```suggestion
The coordinate system is in line with ASAM OpenDRIVE, ASAM OpenSCENARIO, and ASAM OSI.
```

File: content/geometry/general-structure/General_structure.adoc [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+////
+Origins of the coordinate frames for specific object classes are under
+discussion.
+////
+
+== Naming conventions
+The following naming conventions apply to material files:
+
+* Use prefix in the 3D model file name to show that it is according to standard
```
Comment (mettelilienthal) [inline content/geometry/general-structure/General_structure.adoc:23]:
```suggestion
* Use prefixes in the 3D model file name to show that it is up to standard
```

**Review by ACMackenthun — APPROVED** [review]

File: content/geometry/general-structure/General_structure.adoc [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+* Material mapping (with possibility to easily change assignment)
+* Optional link to another asset file to facilitate instancing (same asset file for different 3D models)
+* Semantic information (labels) for sub-meshes ?
+* Description of coordinates, pivot points and so on instead of hierarchy in 3D model file ?
+* Assigning data streams (for example, wheel rotation) to certain sub-meshes ?
+
+////
+TODO: Add short descriptions and references
+////
+
```
Comment (mettelilienthal) [inline content/geometry/general-structure/General_structure.adoc:70]:
```suggestion
Suggestion: Add section for material parameter files analogue to asset file 
```

File: content/geometry/general-structure/File_format_support.adoc [hunk 4]
```diff
@@ -0,0 +1,16 @@
+= File format
+The ASAM OpenMATERIAL standard supports the following file formats:
+* JSON for asset file, which is used for mapping ASAM OpenMATERIAL based on node, mesh, or material names
```
Comment (mettelilienthal) [inline content/geometry/general-structure/File_format_support.adoc:3]:
```suggestion
* JSON for the 3D asset file, which is used for mapping ASAM OpenMATERIAL based on node, mesh, or material names (see also [link to 3D asset in Terms and definitions])
```

File: content/geometry/general-structure/General_structure.adoc [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+Origins of the coordinate frames for specific object classes are under
+discussion.
+////
+
+== Naming conventions
+The following naming conventions apply to ASAM OpenMATERIAL files:
+
+* The name of the 3D model file shall have the prefix `omg_` to indicate that the file
+complies with the ASAM OpenMATERIAL standard.
+* The 3D model file and the related asset file shall have the same base name.
```
Comment (mettelilienthal) [inline content/geometry/general-structure/General_structure.adoc:33]:
```suggestion
* The 3D model file and the related 3D asset file shall have the same base name.
```

File: content/geometry/general-structure/General_structure.adoc [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+*Example*
+If the name of the 3D model file is `omg_my-model.gltf`, then the
+asset file has name `omg_my-model.xoma` (for material assets) or
+`omg_my-model.xomg`(for material geometries).
+
+////
+Putting more information in file name to be discussed for specific object classes.
+////
+
+== Asset file
```
Comment (mettelilienthal) [inline content/geometry/general-structure/General_structure.adoc:44]:
```suggestion
== 3D asset file
```

File: content/geometry/general-structure/General_structure.adoc [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+The asset file is used for mapping ASAM OpenMATERIAL objects based on node, mesh,
```
Comment (mettelilienthal) [inline content/geometry/general-structure/General_structure.adoc:45]:
```suggestion
The 3D asset file is used for mapping ASAM OpenMATERIAL objects based on node, mesh,
```

File: content/geometry/general-structure/General_structure.adoc [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+or material names. It is stored in JSON format.
+
+The asset file either provides material assets or material geometries. The extension of
```
Comment (mettelilienthal) [inline content/geometry/general-structure/General_structure.adoc:48]:
```suggestion
The 3D asset file provides either material assets or material geometries. The extension of
```

File: content/geometry/general-structure/General_structure.adoc [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+an asset file indicates the content type provided:
```
Comment (mettelilienthal) [inline content/geometry/general-structure/General_structure.adoc:49]:
```suggestion
a 3D asset file indicates the content type provided:
```

File: content/geometry/general-structure/General_structure.adoc [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+* `xoma` for material assets. Example: `omg_my-model.xoma`.
+* `xomg` for material geometries. Example: `omg_my-model.xomg`.
+
+////
+Miro board mentions that the extensions `xoma` and `xomg` are "different
+from material files". Clarify which terms to use for
+different files (open material geometry file = 3D model file?).
+////
+
+Each asset file contains the following information:
```
Comment (mettelilienthal) [inline content/geometry/general-structure/General_structure.adoc:59]:
```suggestion
Each 3D asset file contains the following information:
```

File: content/geometry/general-structure/General_structure.adoc [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+Putting more information in file name to be discussed for specific object classes.
+////
+
+== 3D asset file
+The 3D asset file is used for mapping ASAM OpenMATERIAL objects based on node, mesh,
+or material names. It is stored in JSON format.
+
+The 3D asset file provides either material assets or material geometries. The extension of
+a 3D asset file indicates the content type provided:
+* `xoma` for material assets. Example: `omg_my-model.xoma`.
```
[the remark this answers is not in the document]
Comment (FKlopfer) [reply content/geometry/general-structure/General_structure.adoc:50]: Yes, there is an empty line needed before a list in AsciiDoc. I will correct this.

File: content/geometry/general-structure-index.adoc [hunk 12]
```diff
@@ -0,0 +1,12 @@
+= General structure
```
[the remark this answers is not in the document]
Comment (FKlopfer) [reply content/geometry/general-structure-index.adoc]: Renamed the chapter to "General".

File: content/geometry/general-structure/General_structure.adoc [hunk 13]
```diff
@@ -0,0 +1,89 @@
+= General structure
+
+== Dimensions
+The unit used for object dimensions is meters.
+
+== Coordinate systems
+The local coordinate system is a right-handed coordinate system according to
+ISO 8855 with the axes pointing to the directions as follows. For a non-rotated
```
[the remark this answers is not in the document]
Comment (FKlopfer) [reply content/geometry/general-structure/General_structure.adoc]: Rephrased: "The local coordinate system is a right-handed coordinate system according to
ISO 8855 with the axes pointing to the directions mentioned below. For a non-rotated
coordinate system, the following applies:"

File: content/geometry/general-structure/General_structure.adoc [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+
+////
+Putting more information in file name to be discussed for specific object classes.
+////
+
+== 3D asset file
+The 3D asset file is used for mapping ASAM OpenMATERIAL objects based on node, mesh,
+or material names. It is stored in JSON format.
+
+The 3D asset file provides either material assets or material geometries. The extension of
```
[the remark this answers is not in the document]
Comment (FKlopfer) [reply content/geometry/general-structure/General_structure.adoc]: Sentence is rephrased.

File: content/geometry/general-structure/General_structure.adoc [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+* Material mapping (with possibility to easily change assignment)
+* Optional link to another asset file to facilitate instancing (same asset file for different 3D models)
+* Semantic information (labels) for sub-meshes ?
+* Description of coordinates, pivot points and so on instead of hierarchy in 3D model file ?
+* Assigning data streams (for example, wheel rotation) to certain sub-meshes ?
+
+////
+TODO: Add short descriptions and references
+////
+Suggestion: Add section for material parameter files analogue to asset file
```
[the remark this answers is not in the document]
Comment (FKlopfer) [reply content/geometry/general-structure/General_structure.adoc]: Deleted the sentence.

File: content/geometry/general-structure/General_structure.adoc [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+
+The 3D asset file provides metadata and material mapping to the geometry. The extension of
+a 3D asset file indicates the content type provided:
+
+* `xoma` for material assets. Example: `omg_my-model.xoma`.
+* `xomg` for material geometries. Example: `omg_my-model.xomg`.
+
+NOTE: Miro board mentions that the extensions `xoma` and `xomg` are "different
+from material files". Clarify which terms to use for
+different files (open material geometry file = 3D model file?).
```
Comment (ClemensLinnhoff) [inline content/geometry/general-structure/General_structure.adoc:54]: xoma and xomg were just two suggestions for the OpenMATERIAL asset file. I would suggest using `xoma` for now. This is also what I used in the examples. The material parameters will have a different file extension. In the preliminary examples I used `xomp` for "open material properties"

File: content/geometry/general-structure-index.adoc (hunk 12 above)
[the remark this answers is not in the document]
Comment (ClemensLinnhoff) [reply content/geometry/general-structure-index.adoc]: But the file name is sill "General_structure.adoc". And there is also a general-structure-index.adoc as well as the folder name called "general-structure". Can we remove the folder and put everything in content/geometry directly? Then the index file can just be named "geometry_index.adoc"

File: content/geometry/general-structure/General_structure.adoc (hunk 11 above)
[the remark this answers is not in the document]
Comment (ClemensLinnhoff) [reply content/geometry/general-structure/General_structure.adoc:49]: I will rewrite this section.

**Comment by ClemensLinnhoff** [comment]
I don't know, why the DCO fails as I have signed all my commits. Anyone have an idea?

**Comment by ClemensLinnhoff** [comment]
> I don't know, why the DCO fails as I have signed all my commits. Anyone have an idea?

It appears that previous commits are unsigned:

Commit sha: [cdb9099](https://github.com/asam-ev/OpenMATERIAL/pull/36/commits/cdb9099b511f362740c87449f21809c673e93277), Author: Ann-Cathrin Mackenthun, Committer: Ann-Cathrin Mackenthun; The sign-off is missing.
Commit sha: [ac292a1](https://github.com/asam-ev/OpenMATERIAL/pull/36/commits/ac292a181a3f0e9e5a3e6e9be894bd1ded9662cc), Author: Ann-Cathrin Mackenthun, Committer: Ann-Cathrin Mackenthun; The sign-off is missing.
Commit sha: [ac50a48](https://github.com/asam-ev/OpenMATERIAL/pull/36/commits/ac50a489801b55dbd1ab089066f1f32be670d70e), Author: Ann-Cathrin Mackenthun, Committer: Ann-Cathrin Mackenthun; The sign-off is missing.
Commit sha: [734990d](https://github.com/asam-ev/OpenMATERIAL/pull/36/commits/734990d0dafd453f26386566a8a2cfdc0d1b590d), Author: FabianKlopfer, Committer: FabianKlopfer; Expected "FabianKlopfer [fabian.klopfer@parson-europe.com](mailto:fabian.klopfer@parson-europe.com)", but got "Fabian Klopfer [fabian.klopfer@parson-europe.com](mailto:fabian.klopfer@parson-europe.com)".
Commit sha: [b9347c9](https://github.com/asam-ev/OpenMATERIAL/pull/36/commits/b9347c9665ce161c562cc68b9f2cdb297c0f0f6c), Author: mette.lilienthal, Committer: mette.lilienthal; The sign-off is missing.
Commit sha: [b0568d0](https://github.com/asam-ev/OpenMATERIAL/pull/36/commits/b0568d01a5ad53e21eba49cc422153f2d5d52fec), Author: mettelilienthal, Committer: GitHub; The sign-off is missing.
Commit sha: [b275b77](https://github.com/asam-ev/OpenMATERIAL/pull/36/commits/b275b7771c9168151bc83a7c5097997e504ea10e), Author: mettelilienthal, Committer: GitHub; The sign-off is missing.

@FKlopfer could you fix that?

File: content/geometry/general-structure-index.adoc (hunk 12 above)
[the remark this answers is not in the document]
Comment (FKlopfer) [reply content/geometry/general-structure-index.adoc]: - Renamed General_structure.adoc --> General.adoc
- Deleted folder "general-structure" and copy all files to the level above
- Once all PR are merged (and I have a better overview where which file is located), I will create a consolidated geometry-index.adoc file for the whole geometry chapter

**Comment by ClemensLinnhoff** [comment]
@FKlopfer after your structure change, the general chapter is now twice in there. Probably because I changed the content of that chapter in my previous commit. Did you do any content changes to General.adoc? If not, you can delete it and rename the remaining general-structure/General_structure.adoc to General.adoc.

File: content/geometry/general-structure-index.adoc (hunk 12 above)
[the remark this answers is not in the document]
Comment (ClemensLinnhoff) [reply content/geometry/general-structure-index.adoc]: (check my comment below on the new structure)

**Comment by FKlopfer** [comment]
> > I don't know, why the DCO fails as I have signed all my commits. Anyone have an idea?
> 
> It appears that previous commits are unsigned:
> 
> Commit sha: [cdb9099](https://github.com/asam-ev/OpenMATERIAL/pull/36/commits/cdb9099b511f362740c87449f21809c673e93277), Author: Ann-Cathrin Mackenthun, Committer: Ann-Cathrin Mackenthun; The sign-off is missing. Commit sha: [ac292a1](https://github.com/asam-ev/OpenMATERIAL/pull/36/commits/ac292a181a3f0e9e5a3e6e9be894bd1ded9662cc), Author: Ann-Cathrin Mackenthun, Committer: Ann-Cathrin Mackenthun; The sign-off is missing. Commit sha: [ac50a48](https://github.com/asam-ev/OpenMATERIAL/pull/36/commits/ac50a489801b55dbd1ab089066f1f32be670d70e), Author: Ann-Cathrin Mackenthun, Committer: Ann-Cathrin Mackenthun; The sign-off is missing. Commit sha: [734990d](https://github.com/asam-ev/OpenMATERIAL/pull/36/commits/734990d0dafd453f26386566a8a2cfdc0d1b590d), Author: FabianKlopfer, Committer: FabianKlopfer; Expected "FabianKlopfer [fabian.klopfer@parson-europe.com](mailto:fabian.klopfer@parson-europe.com)", but got "Fabian Klopfer [fabian.klopfer@parson-europe.com](mailto:fabian.klopfer@parson-europe.com)". Commit sha: [b9347c9](https://github.com/asam-ev/OpenMATERIAL/pull/36/commits/b9347c9665ce161c562cc68b9f2cdb297c0f0f6c), Author: mette.lilienthal, Committer: mette.lilienthal; The sign-off is missing. Commit sha: [b0568d0](https://github.com/asam-ev/OpenMATERIAL/pull/36/commits/b0568d01a5ad53e21eba49cc422153f2d5d52fec), Author: mettelilienthal, Committer: GitHub; The sign-off is missing. Commit sha: [b275b77](https://github.com/asam-ev/OpenMATERIAL/pull/36/commits/b275b7771c9168151bc83a7c5097997e504ea10e), Author: mettelilienthal, Committer: GitHub; The sign-off is missing.
> 
> @FKlopfer could you fix that?

It seems the check fails if only one previous DCO is missing. It is possible to manually approve the DCO by clicking on "Checks" in GitHub and then "Set DCO to pass".

**Comment by FKlopfer** [comment]
> @FKlopfer after your structure change, the general chapter is now twice in there. Probably because I changed the content of that chapter in my previous commit. Did you do any content changes to General.adoc? If not, you can delete it and rename the remaining general-structure/General_structure.adoc to General.adoc.

Yes, we shouldn't have worked in parallel on that. I did not change the content of the file and did as you proposed. There is only one "General.adoc" file now.

**Review by ClemensLinnhoff — APPROVED** [review]

**Comment by ClemensLinnhoff** [comment]
@LudwigFriedmann could you review this PR? If you approve, we can decide to merge in the CCB tomorrow.

**Comment by ClemensLinnhoff** [comment]
Ah I just noticed something, @FKlopfer : The general-structure-index.adoc needs to be added to the content/index.adoc file, correct? Otherwise the chapter will not be compiled into the html documentation.

**Comment by FKlopfer** [comment]
> Ah I just noticed something, @FKlopfer : The general-structure-index.adoc needs to be added to the content/index.adoc file, correct? Otherwise the chapter will not be compiled into the html documentation.

This is what I meant with one of my previous comments about consolidating the index file. When the content of all open PR are merged, I would create a new PR and insert the links to a consolidated index file there. This PR can possibly be merged quite quickly after that.

**Comment by ClemensLinnhoff** [comment]
> > Ah I just noticed something, @FKlopfer : The general-structure-index.adoc needs to be added to the content/index.adoc file, correct? Otherwise the chapter will not be compiled into the html documentation.
> 
> This is what I meant with one of my previous comments about consolidating the index file. When the content of all open PR are merged, I would create a new PR and insert the links to a consolidated index file there. This PR can possibly be merged quite quickly after that.

But if we create a working structure here already, we can adapt the other PRs to fit to that structure without the need for an additional PR.

**Comment by ClemensLinnhoff** [comment]
Add introduction chapter.

**Comment by ClemensLinnhoff** [comment]
CCB decision 02.08.2024: Merge after indexing and adding introduction chapter.

**Comment by FKlopfer** [comment]
Added introduction chapter and reference all chapters in index.adoc and nav.adoc.

**Pull request merged**